### PR TITLE
Enable HTTP probe by default

### LINKED ISSUE #466: High CPU reported from default execProbes
## Expected Behaviour

Low ambient CPU usage.

## Current Behaviour

High CPU reported from default execProbes for workshop students due to the current helm chart defaults. They are cautiously set to "httpProbe: false" for backwards compatibility.

## Possible Solution

Set httpProbe: true for all functions as a default.

## Steps to Reproduce (for bugs)

1. Deploy 10 functions with execProbes
2. Observe CPU over 1 hour
3. Do the same but with httpProbe

## Context

Reported by @andyjeffries at Civo Cloud after doing labs 2-3 of the workshop.

## Your Environment

Kubernetes 1.14 with k3s from Rancher Labs

2 vCPU VM on OpenStack with 4GB RAM.

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis <alexellis2@gmail.com>

## Description

Enable HTTP probe by default

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Fixes #466

This will reduce user-facing CPU usage.

Change also increases function polling time to 2s to reduce
the CPU overhead. Users can set this to 1s for faster cold-starts.

## How Has This Been Tested?

Tested with common function store images. Some older images will need to be rebuilt by the user.

I would recommend that we introduce a flag that can set the probe type on a per-function basis for backwards-compatibility.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: chart/openfaas/README.md [hunk 1]
```diff
@@ -70,12 +70,44 @@ helm repo update \
...
+3) Tune the readinessProbes to be aggressively low values. This will reduce the cold-start at the cost of more `kubelet` CPU usage
+
+To achieve around 1s coldstart, set `values.yaml`:
+
+```yaml
+faasnetes:
+
+# redacted
+  readinessProbe:
+    initialDelaySeconds: 1
```
Comment (stefanprodan) [inline chart/openfaas/README.md]: The initial delay can be set to zero see https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/v1/defaults_test.go#L1611

Comment (alexellis) [reply chart/openfaas/README.md]: I have a recollection about setting one or more of these values to zero and them being automatically changed back to "1". I can't remember which one, but I thought it was this one.

Comment (alexellis) [reply chart/openfaas/README.md]: I've set it to zero on your recommendation in the "tuning" section.

**Comment by alexellis** [comment]
Thanks both :+1:

**Pull request merged**